examples/digital_fingerprinting/production/morpheus/dfp/modules/dfp_rolling_window.py

```diff
@@ -0,0 +1,164 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+import typing
+from contextlib import contextmanager
+
+import mrc
+import pandas as pd
+from dfp.utils.cached_user_window import CachedUserWindow
+from dfp.utils.logging_timer import log_time
+from mrc.core import operators as ops
+
+from morpheus.utils.module_utils import get_module_config
+from morpheus.utils.module_utils import register_module
+
+from ..messages.multi_dfp_message import DFPMessageMeta
+from ..messages.multi_dfp_message import MultiDFPMessage
+
+logger = logging.getLogger(f"morpheus.{__name__}")
+
+
+@register_module("DFPRollingWindow", "morpheus_modules")
+def dfp_rolling_window(builder: mrc.Builder):
+    """
+    This module function establishes a rolling window to maintain history.
+
+    Parameters
+    ----------
+    builder : mrc.Builder
+        Pipeline budler instance.
+    """
+
+    module_id = "DFPRollingWindow"
+
+    config = get_module_config(module_id, builder)
+
+    timestamp_column_name = config.get("timestamp_column_name", None)
+    min_history = config.get("min_history", None)
+    max_history = config.get("max_history", None)
+    min_increment = config.get("min_increment", None)
+    cache_dir = config.get("cache_dir", None)
+
+    cache_dir = os.path.join(cache_dir, "rolling-user-data")
+
+    user_cache_map: typing.Dict[str, CachedUserWindow] = {}
+
+    @contextmanager
+    def get_user_cache(user_id: str):
+
+        # Determine cache location
+        cache_location = os.path.join(cache_dir, f"{user_id}.pkl")
+
+        user_cache = None
+
+        user_cache = user_cache_map.get(user_id, None)
+
+        if (user_cache is None):
+            user_cache = CachedUserWindow(user_id=user_id,
+                                          cache_location=cache_location,
+                                          timestamp_column=timestamp_column_name)
+
+            user_cache_map[user_id] = user_cache
+
+        yield user_cache
+
+        # # When it returns, make sure to save
+        # user_cache.save()
```

Comment:
Is this supposed to be commented out?

Reply:
It's commented in the original [file](https://github.com/bsuryadevara/Morpheus/blob/62149a773e48468e5a561b83a2c0c20bfff249b4/examples/digital_fingerprinting/production/morpheus/dfp/stages/dfp_rolling_window_stage.py#L115)